### PR TITLE
Example for inference using a single TPU

### PR DESCRIPTION
**Pull request by xiangshen-dk**
Added an example (MINST) which is used for a quickstart tutorial.

### CODE REVIEW

**Comment by alizaidis** [comment]
@ahg-g Any chance we can get a review on this example showing how to train and do batch inference using MNIST on a single TPU VM? Much appreciated!

File: gke-tpu-examples/training/mnist-single-tpu/src/gke-config/standard-tensorflow-bash-v4.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+apiVersion: v1
+kind: Pod
```
Comment (ahg-g) [inline gke-tpu-examples/training/mnist-single-tpu/src/gke-config/standard-tensorflow-bash-v4.yaml:2]: We shouldn't recommend creating plain pods, why not a job?

File: gke-tpu-examples/training/mnist-single-tpu/src/gke-config/standard-tf-mnist-batch-predict-v4.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      nodeSelector:
+        cloud.google.com/gke-tpu-topology: 2x2x1  # target topology
+        cloud.google.com/gke-tpu-accelerator: tpu-v4-podslice
+      containers:
+      - name: tensorflow
+        image: tensorflow/tensorflow:2.14.0
+        securityContext:
+          privileged: true
+        command: ["/bin/bash", "-c", "--"]
+        args: ["cd /data/tensorflow-mnist-example; pip install -r requirements.txt; python tensorflow_mnist_batch_predict.py"]
```
Comment (ahg-g) [inline gke-tpu-examples/training/mnist-single-tpu/src/gke-config/standard-tf-mnist-batch-predict-v4.yaml:21]: where do we create the container that includes the `tensorflow_mnist_batch_predict.py` script?

File: gke-tpu-examples/training/mnist-single-tpu/src/gke-config/standard-tensorflow-bash-v4.yaml (hunk 1 above)
Comment (xiangshen-dk) [reply gke-tpu-examples/training/mnist-single-tpu/src/gke-config/standard-tensorflow-bash-v4.yaml:2]: That's the manifest to create a pod so the user can have a shell session to test the volume and verify the TPU attachment. The real training and inferencing are using jobs.

File: gke-tpu-examples/training/mnist-single-tpu/src/gke-config/standard-tf-mnist-batch-predict-v4.yaml (hunk 2 above)
Comment (xiangshen-dk) [reply gke-tpu-examples/training/mnist-single-tpu/src/gke-config/standard-tf-mnist-batch-predict-v4.yaml:21]: It should be on the TPU node based on the node selector.

Comment (ahg-g) [reply gke-tpu-examples/training/mnist-single-tpu/src/gke-config/standard-tf-mnist-batch-predict-v4.yaml:21]: oh, the script is accessed from gcs; ok, this is a slightly strange setup, usually we put it inside the container

**Review by ahg-g — APPROVED** [review]

**Pull request merged**